File: samples/bot-task-modules/dotnet/bot-task-modules/Models/TaskModuleConfig.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

namespace BotTaskModules.Models;

// UI settings for task modules.
public class UISettings
{
    public int Width { get; set; }
    public int Height { get; set; }
    public string Title { get; set; }
    public string Id { get; set; }
    public string ButtonTitle { get; set; }

    public UISettings(int width, int height, string title, string id, string buttonTitle)
    {
        Width = width;
        Height = height;
        Title = title;
        Id = id;
        ButtonTitle = buttonTitle;
    }
}

// Task module identifiers and settings.
public static class TaskModuleIds
{
    public const string CustomForm = "CustomForm";
    public const string AdaptiveCard = "AdaptiveCard";
}

// Task module UI constants.
public static class TaskModuleUIConstants
{
    public static readonly UISettings CustomForm = new(510, 450, "Custom Form", TaskModuleIds.CustomForm, "Custom Form");
    public static readonly UISettings AdaptiveCard = new(400, 200, "Adaptive Card: Inputs", TaskModuleIds.AdaptiveCard, "Adaptive Card");
}

File: samples/bot-task-modules/dotnet/bot-task-modules/bot-task-modules_python/task_module_config.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License.


class UISettings:
    """UI settings for task modules."""
    def __init__(self, width: int, height: int, title: str, id: str, button_title: str):
        self.width = width
        self.height = height
        self.title = title
        self.id = id
        self.button_title = button_title


class TaskModuleIds:
    """Task module identifiers and settings."""
    CUSTOM_FORM = UISettings(510, 450, "Custom Form", "CustomForm", "Custom Form")
    ADAPTIVE_CARD = UISettings(400, 200, "Adaptive Card: Inputs", "AdaptiveCard", "Adaptive Card")
